Source/MatcherFactory.cs

```diff
@@ -94,28 +94,15 @@ public static IMatcher CreateMatcher(Expression expression, bool isParams)
 					}
 				}
 
-				var attr = call.Method.GetCustomAttribute<AdvancedMatcherAttribute>(true);
 #pragma warning disable 618
-				var staticMatcherMethodAttr = call.Method.GetCustomAttribute<MatcherAttribute>(true);
-#pragma warning restore 618
-
-				if (attr != null)
+				if (call.Method.IsDefined(typeof(MatcherAttribute), true))
 				{
-					var matcher = attr.CreateMatcher();
-					matcher.Initialize(originalExpression);
-					return matcher;
-				}
-				else if (staticMatcherMethodAttr != null)
-				{
-					var matcher = new MatcherAttributeMatcher();
-					matcher.Initialize(originalExpression);
-					return matcher;
+					return new MatcherAttributeMatcher(call);
```

Comment:
Passing `call` instead of `originalExpression` is actually a small change, as the latter has had any type conversion node removed and the former hasn't. But it would seem more correct since code inside `MatcherAttributeMatcher` silently assumes that it's given a `MethodCallExpression` (and not a conversion node).